src/utils/createDispatcher.js

```diff
@@ -0,0 +1,38 @@
+import getContextValue from './getContextValue';
+
+const createDispatcher = (context) => ({
+  readContext: (suppliedContext) => {
+    return getContextValue(context, suppliedContext);
+  },
+  useContext: (suppliedContext) => {
+    return getContextValue(context, suppliedContext);
+  },
+  useEffect: () => {},
+  useState: (initial) => [typeof initial === 'function' ? initial() : initial],
+  useReducer: (reducer, initArg, initializer) => [
+    !initializer ? initArg : initializer(initArg),
+  ],
+  useMemo: (computeFunction) => computeFunction(),
+  useCallback: (callbackFunction) => () => callbackFunction(),
+  useRef: (initial) => ({ current: initial }),
+  useImperativeHandle: () => {},
+  useLayoutEffect: () => {},
+  useInsertionEffect: () => {},
+  useDebugValue: () => {},
+  useDeferredValue: (value) => value,
+  useTransition: () => [false, () => {}],
+  /*
+    Ideally the useId hook should work in prepare(), however the logic used by
+    react to generate ids is dependent on the full component tree, making it
+    difficult to imitate. The ids must also be consistent between client and
+    server, so using a different implementation is not an option.
+
+    In most cases the useId hook should just be used for generating dom element
+    ids, which should be irrelevant to react-prepare.
+   */
+  useId: () => undefined,
```

Comment:
I don't see any possible usecase for this to be relevant to `react-prepare` tbh (why wants a random react it in a data fetching function???), so I think it's perfectly fine to ignore it :+1:

Reply:
Agreed. Should I leave the comment?

Reply:
I think the comment is a bit confusing and might be misleading, so I say remove it. You can maybe leave the last sentence.